gravitee-apim-cypress/cypress/commands/management/api-management-commands.ts

```diff
@@ -13,33 +13,100 @@
+    failOnStatusCode,
   });
 }
 
-export function publishApi(auth: BasicAuthentication, apiId: string) {
+export function publishApi(auth: BasicAuthentication, createdApi: Api, failOnStatusCode = false) {
+  const apiToPublish = {
+    ...createdApi,
+    lifecycle_state: ApiLifecycleState.PUBLISHED,
+  };
+  delete apiToPublish.id;
+  delete apiToPublish.state;
+  delete apiToPublish.created_at;
+  delete apiToPublish.updated_at;
+  delete apiToPublish.owner;
+  delete apiToPublish.context_path;
   return cy.request({
     method: 'PUT',
-    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}`,
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${createdApi.id}`,
+    body: apiToPublish,
     auth,
-    failOnStatusCode: false,
+    failOnStatusCode,
   });
 }
 
-export function deleteApi(auth: BasicAuthentication, apiId: string) {
+export function deleteApi(auth: BasicAuthentication, apiId: string, failOnStatusCode = false) {
   return cy.request({
     method: 'DELETE',
     url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}`,
     auth,
-    failOnStatusCode: false,
+    failOnStatusCode,
+  });
+}
+
+export function deployApi(auth: BasicAuthentication, apiId: string, failOnStatusCode = false) {
+  return cy.request({
+    method: 'POST',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}/deploy`,
+    auth,
+    failOnStatusCode,
+  });
+}
+
+export function startApi(auth: BasicAuthentication, apiId: string, failOnStatusCode = false) {
+  return cy.request({
+    method: 'POST',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}?action=START`,
+    auth,
+    failOnStatusCode,
+  });
+}
+
+export function stopApi(auth: BasicAuthentication, apiId: string, failOnStatusCode = false) {
+  return cy.request({
+    method: 'POST',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}?action=STOP`,
+    auth,
+    failOnStatusCode,
+  });
+}
+
+export function createPlan(auth: BasicAuthentication, apiId: string, body: Partial<NewPlanEntity>, failOnStatusCode = false) {
+  return cy.request({
+    method: 'POST',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}/plans`,
+    auth,
+    body,
+    failOnStatusCode,
+  });
+}
+
+export function deletePlan(auth: BasicAuthentication, apiId: string, planId: string, failOnStatusCode = false) {
+  return cy.request({
+    method: 'DELETE',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}/plans/${planId}`,
+    auth,
+    failOnStatusCode,
+  });
+}
+
+export function publishPlan(auth: BasicAuthentication, apiId: string, planId: string, failOnStatusCode = false) {
+  return cy.request({
+    method: 'POST',
+    url: `${Cypress.config().baseUrl}${Cypress.env('managementApi')}/apis/${apiId}/plans/${planId}/_publish`,
+    auth,
+    failOnStatusCode,
```

Comment:
please avoid mixing multiple concepts
why not creating a `api-plan-management-commands.ts` ?

Reply:
Yes, absolutely. Ok, will do that.